test/funder.js

```diff
@@ -0,0 +1,328 @@
+ */
+
+
+/**
+ * @typedef {Object} FundingDetails
+ *
+ * @property {FundingReceipts} Receipts of all the faucet.
+ * @property {FundingTransactionHashes} transactionHashes of all the faucet.
+ */
+
+/**
+ * @typedef {Object} ERC20RefundTransaction
+ * @property {string}  type Type of fund.
+ * @property {string} faucetAddress Address of faucet.
+ * @property {string} tokenAddress Address of ERC20 contract.
+ */
+
+/**
+ * @typedef {Object} BaseTokenRefundTransaction
+ * @property {string}  type Type of fund.
+ * @property {string} faucetAddress Address of faucet.
+ */
+
+/**
+ *
+ * @typedef {Object} RefundTransactionDetails
+ * @property {ERC20RefundTransaction} originTransactions Return to faucet
+ *                                                        transaction detail for
+ *                                                        origin chain.
+ * @property {BaseTokenRefundTransaction} auxiliaryTransactions Return to faucet
+ *                                                        transaction detail for
+ *                                                        auxiliary chain.
+ * @property {BaseTokenRefundTransaction} ropstenTransactions Return to faucet
+ *                                                        transaction detail for
+ *                                                        the ropsten.
+ */
+
+const axios = require('axios');
+const Web3 = require('web3');
+const Mosaic = require('@openst/mosaic.js');
+
+const Account = require('../src/account');
+const { MOSAIC_FAUCET_URL, ROPSTEN_FAUCET_URL, ROPSTEN_REFUND_ADDRESS } = require('./constants');
+const shared = require('./shared');
+
+const DEFAULT_PASSWORD = 'JLP_PASSWORD';
+const ORIGIN_CHAIN_ID = 3;
+const AUXILIARY_CHAIN_ID = 200;
+
+const { BN } = Web3.utils;
+
+/**
+ * This creates an account for origin chain and save it in shared object.
+ * @param {string} name Account identifier.
+ * @param {We3} web3 origin Web3 instance.
+ */
+const addOriginAccount = async (name, web3) => {
+  const accountAddress = await Account.create(name, web3, DEFAULT_PASSWORD);
+  shared.accounts.origin[name] = {
+    name,
+    chainId: ORIGIN_CHAIN_ID,
+    address: accountAddress,
+  };
+};
+
+/**
+ * This creates an account for auxiliary chain and save it in shared object.
+ * @param {string} name Account identifier.
+ * @param {We3} web3 auxiliary Web3 instance.
+ */
+const addAuxiliaryAccount = async (name, web3) => {
+  const accountAddress = await Account.create(name, web3, DEFAULT_PASSWORD);
+  shared.accounts.auxiliary[name] = {
+    name,
+    chainId: AUXILIARY_CHAIN_ID,
+    address: accountAddress,
+  };
+};
+
+/**
+ * This funds an account from mosaic faucet.
+ * @param {string} address Beneficiary address where funds will be added;
+ * @param {string} chainId Chain Id of chain where beneficiary will get fund.
+ * @return {Promise<string>} Transaction hash returned by faucet.
+ */
+const fundAccountFromMosaicFaucet = (address, chainId) => new Promise(
```

Comment:
Generally, you almost never need to create your own `new Promise`. Why not `return axios.post...`? You can still use `then` and `catch`.
Same in other functions.

Reply:
If you see, This method doesn't return `response` from the request. It returns `response.data`. 
I could remove `Promise`, but need to add await for the response. 

From my understanding adding `async` and explicitly adding promise is exactly the same.